### ISSUE TITLE
Lots of Authenticated Client Devices

### ISSUE THREAD

**Issue by developedsoftware** [dgreif/ring#1059]
# Bug Report

### Describe the Bug
Whenever you run an api call a new client device appears in your list of authorised devices. Even though all the calls use the same access token. 

![6B6670F1-D874-436E-828A-9D718090BD5B](https://user-images.githubusercontent.com/19648131/192091323-4cff14e9-490a-42d3-9d2e-f926c4c7c0d7.png)

### To Reproduce
 - Create an `ApiClient`
 - Attempt to run any api call
 - See new device added to your list of authorised client devices

### Expected behavior
One device per token ? So just one device in this case ? 

### Screenshots/Logs
 If applicable, add screenshots and log output to help explain your problem.

### Additional context
Not sure if this is by design or not - but it’s extremely frustrating ! 

### Environment

 - OS: [e.g. Raspbian]
 - Node.js: [e.g. 12.14.1] 
 - NPM: [e.g 6.13.4] 
 - ring-client-api: [e.g 6.0.3]

**Comment by tsightler** [comment]
You need to use a persistent systemId and pass this as part of the ApiClient function to avoid this, otherwise a random system UUID is generated each time a new ApiClient is created and thus a new authorized client is added to the list.

**Comment by tsightler** [comment]
To be completely clear, ring-client-api does attempt to generate a persistent UUID even if you don't pass systemId, if you look at the getHardwareId() function in util, if a systemId is not provided it tries to get a unique hardware UUID from the systeminformation package, but in many cases this either fails to return a valid value, in which case ring-client-api falls back to generating a random value, or, if you are running the code in cases where the runtime environment is different each time (think serverless like running the code on Fargate or even just a case where the code is run in a new container each time), then systeminformation will still return a new UUID each time, thus technically it is a new device.

In summary, if you don't want a new device created on the ring side, then you have to have not just the same token, but the same systemId, and the most sure way to do that is to pass in a persistent systemId as well.  For ring-mqtt I generate my on systemUuid during initial token authentication and store that with the token so it is always the same, vs relying on the runtime environment to stay consistent.

**Comment by developedsoftware** [comment]
That’s great thank you !

I am actually using docker so I can probably ensure the systemid stays the same. Thanks for the info / I should be able to work with that! 

Cheers

**Comment by developedsoftware** [comment]
Not sure if I am being thick here, but I assumed I’d be able to pass in the systemId when creating a new instance of the RingApi ? 

```
const ringApi = new RingApi({
        refreshToken: process.env.token as string,
        debug: false
    });
```

**Comment by tsightler** [comment]
Yes, just pass additional systemId parameter and it will be used to see uuid creation.  You can find examples in the ring-homebridge code also in this repository.

**Comment by developedsoftware** [comment]
This has worked perfectly, thank you!
